tests/testthat/test-dictionary.R

```diff
@@ -254,29 +226,21 @@ test_that("can use matching functions with strings with different encodings", {
 })
 
 test_that("can use matching functions when one string has multiple encodings", {
-  utf8 <- "\u00B0C"
-
-  unknown <- utf8
-  Encoding(unknown) <- "unknown"
+  enc <- encodings()
 
-  latin1 <- iconv(utf8, "UTF-8", "latin1")
-
-  x <- c(utf8, unknown)
+  x <- c(enc$utf8, enc$unknown)
+  latin1 <- enc$latin1
 
   expect_equal(vec_match(x, latin1), c(1L, 1L))
   expect_equal(vec_match(x, latin1), match(x, latin1))
 })
 
 test_that("can use matching functions within the same encoding", {
-  unknown <- "fa\xE7ile"
-
-  latin1 <- unknown
-  Encoding(latin1) <- "latin1"
-
-  utf8 <- enc2utf8(latin1)
-
-  bytes <- unknown
-  Encoding(bytes) <- "bytes"
+  enc <- encodings()
+  utf8 <- enc$utf8
+  unknown <- enc$unknown
+  latin1 <- enc$latin1
+  bytes <- enc$bytes
 
   expect_equal(vec_match(unknown, unknown), 1L)
```

Comment:
Maybe it's worth turning this into a nested for loop to test every combination?

Reply:
Did this in a number of places, simplifying many of the tests